thetis/timeintegrator.py

```diff
@@ -476,8 +475,11 @@ def __init__(self, equation, solution, fields, dt, bnd_conditions=None,
         self.msolution_old = Function(fs, name='dual solution')
         self.rhs_func = Function(fs, name='rhs linear form')
 
-        continuity = element_continuity(fs.ufl_element())
-        self.fs_is_dg = continuity.horizontal == 'dg' and continuity.vertical == 'dg'
+        try:
+            continuity = element_continuity(fs.ufl_element())
+            self.fs_is_dg = continuity.horizontal == 'dg' and continuity.vertical == 'dg'
+        except Exception:
```

Comment:
What is it exactly that throws an `Exception` ? And is catching `Exception` not equivalent to a blank except, since all exceptions are derived (sub-classed) from `Exception` ? If it really is some code that throws a generic Exception (wherever that happens), could we fix that?

Reply:
Blank `except:` has been deemed "bad" by the `flake8` dictators.

Reply:
Yes, just to clarify: I wasn't suggesting changing it to a blank except, just objecting to `except Exception` _because_ it seems equivalent to a blank except (which I agree is usually bad).

Reply:
Thanks for pointing this out. The `try` clause was there to handle cases where `element_continuity` doesn't work - admittedly bad design. I have now extended `element_continuity` function to handle the mixed function space used in the tests, so there's no need for `try` anymore.